docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,383 @@
+>  * [`ONE_CLIENT_CSP_REPORTING_URL`](../api/server/environment-variables#one_client_csp_reporting_url) ⚠️
+
+While `HOLOCRON_MODULE_MAP_URL` is a required environment variable in `production`,
+we will go into greater depths on this variable [in the next section on Holocron runtime.](#holocron-runtime)
+
+>  * [`HOLOCRON_MODULE_MAP_URL`](../api/server/environment-variables#holocron_module_map_url) ⚠️
+
+As we get to production, we can configure a certificate for One App
+to use and the HTTPS `port` provided (no defaults) for One App. In addition,
+the IP address can be set as well (defaults to `0.0.0.0` during production):
+
+>  * [`IP_ADDRESS`](../api/server/environment-variables#ip_address)
+>  * [`HTTPS_PORT`](../api/server/environment-variables#https_port) ⚠️
+>  * [`HTTPS_PRIVATE_KEY_PASS_FILE_PATH`](../api/server/environment-variables#https_private_key_pass_file_path) ⚠️
+>  * [`HTTPS_PRIVATE_KEY_PATH`](../api/server/environment-variables#https_private_key_path) ⚠️
+>  * [`HTTPS_PUBLIC_CERT_CHAIN_PATH`](../api/server/environment-variables#https_public_cert_chain_path) ⚠️
+>  * [`HTTPS_TRUSTED_CA_PATH`](../api/server/environment-variables#https_trusted_ca_path) ⚠️
+
+### Holocron Runtime
+
+Before One App can start rendering HTML documents from the server, the app will
+preload every Holocron module defined in the Holocron module map. This module map
+is comprised of every module to be used by One App, including the root module which
+serves as our entry point.
+[You can read more about the module map and how it is structured.](../api/server/Module-Map-Schema.md)
+
+One App uses the environment variable `HOLOCRON_MODULE_MAP_URL` to configure from where
+the module map is loaded (as `JSON`). In `development`, there is a local CDN ran alongside One App
+to serve local modules and the module map - please note that when running One App directly
+(repository or Docker image) or using `one-app-runner` will automatically configure this for us.
+When in `production` mode, this variable is required.
+
+>  * [`HOLOCRON_MODULE_MAP_URL`](../api/server/environment-variables#holocron_module_map_url) ⚠️
+
+You can further configure the behavior of Holocron with the Server Settings below:
+
+>  * [`HOLOCRON_SERVER_MAX_MODULES_RETRY`](../api/server/environment-variables#holocron_server_max_modules_retry)
+>  * [`HOLOCRON_SERVER_MAX_SIM_MODULES_FETCH`](../api/server/environment-variables#holocron_server_max_sim_modules_fetch)
+>  * [`ONE_MAP_POLLING_MAX`](../api/server/environment-variables#one_map_polling_max)
+>  * [`ONE_MAP_POLLING_MIN`](../api/server/environment-variables#one_map_polling_min)
+
+One App will first fetch the Holocron module map using the provided URL
+and proceeds to load every module defined in the module map into memory.
+If a module failed to load for any reason or when validating, One App will
+log the error and recover gracefully, adding the module to a `block` list until
```

Comment:
When this happens during the "start" phase one app will not recover and the server will fail to start. Also not all conditions result in the module being added to the block list.

Reply:
Good catch, will update.

Reply:
Updated in (7e618c0)